modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/odbclient/ODBClient.scala

```diff
@@ -0,0 +1,38 @@
+package edu.gemini.seqexec.web.server.odbclient
+
+import edu.gemini.seqexec.model.Model.SequenceId
+import edu.gemini.spModel.core.SPProgramID
+import knobs.Config
+import org.http4s.client.blaze._
+import org.http4s.{Uri, scalaxml}
+
+import scala.xml.Elem
+import scalaz.Kleisli
+import scalaz.concurrent.Task
+
+case class ODBClientConfig(odbHost: String)
+
+case class ODBClient(config: ODBClientConfig) {
+  val httpClient = PooledHttp1Client()
+  // Entity Decoder for xml
+  implicit val decoder = scalaxml.xml()
+  def observationTitle(id: SPProgramID, obsId: SequenceId): Task[String] = {
+    val baseUri = s"${config.odbHost}/odbbrowser/observations"
+    val target = Uri.fromString(baseUri).toOption.get +?("programReference", id.stringValue)
+    httpClient.expect[Elem](target).map { xml =>
+      for {
+        x <- xml \\ "observations" \ "observation"
```

Comment:
These are from `scala-xml` which is part of core scala since forever. It basically takes the root, finds any node `observations` then one node down `observation`, and filters on the id